cni/pkg/ambient/net.go

```diff
@@ -238,20 +240,15 @@ func buildEbpfArgsByIP(ip string, isZtunnel, isRemove bool) (*ebpf.RedirectArgs,
 }
 
 // GetHostIPByRoute get the automatically chosen host ip to the Pod's CIDR
-func GetHostIPByRoute(kubeClient kubernetes.Interface) (string, error) {
+func GetHostIPByRoute(podLister listerv1.PodLister) (string, error) {
 	// We assume per node POD's CIDR is the same block, so the route to the POD
 	// from host should be "same". Otherwise, there may multiple host IPs will be
 	// used as source to dial to PODs.
-	pods, err := kubeClient.CoreV1().Pods(metav1.NamespaceAll).List(
-		context.TODO(),
-		metav1.ListOptions{
-			LabelSelector: "app=ztunnel",
-			FieldSelector: "spec.nodeName=" + NodeName,
-		})
+	pods, err := podLister.List(labels.Set{"app": "ztunnel"}.AsSelector())
```

Comment:
do not equal

Reply:
Yes, don't we need the nodeName selector?

Reply:
Initially, I think while this piece of code called, we should have an active ztunnel, thus stick to ztunnel pod here.

Reply:
this lister should share the podInformer's  Indexer, which has already initiated with the nodeName selector. refer to

[podInformer](https://github.com/istio/istio/blob/af83bc1e0fdffe6bbd3317de81de36e33d53a9e4/cni/pkg/ambient/informers.go#L48)

Reply:
nice